### PR TITLE
RANCHER-927 ready state for PR review by peers.

### PR DESCRIPTION
**Pull request by eldiiar-duishenaliev**

### CODE REVIEW

File: vars/folioJiraFlow.groovy [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    ]
+    String updateUrl = Constants.FOLIO_JIRA_ISSUE_URL + "${ticket}/transitions"
+    teamsInfo.each { -> team
+      String body
+      def ticketNumbers = new JsonSlurperClassic().parseText(searchForExistingJiraTickets(jiraFlowJQL.getOpenTickets(summary, "${team.keySet().first()}")))
+      if (ticketNumbers["total"] != 0) {
+        ticketNumbers["issues"].each { ticket ->
+          switch (ticket["fields"]["status"]["name"]) {
+            case "Open":
+              new Common(this, "https://fakeUrl").logger.info("Similar ticket already exists, summary: ${summary}")
```
Comment (OHaimanov) [inline vars/folioJiraFlow.groovy:121]: It is better to initialize the Common class at the start of the function
Also if you need a logger, is better to initialize the Logger class from the utilities package.

File: resources/jiraFlow/jiraFlowJQL.groovy [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+summary ~ ${summary} AND "Development Team" = ${team} AND status in (Active,"In progress","In Review","In Code Review") AND project = ${Constants.DM_JIRA_PROJECT}"
+  """
+  return jql
+}
+
+String getTicketInfo(String ticketNumber){
+  String jql = """
+issue=${ticketNumber}
+"""
+}
```
[the remark this answers is not in the document]
Comment (eldiiar-duishenaliev) [reply resources/jiraFlow/jiraFlowJQL.groovy:19]: Thanks Sasha, indeed, it missed.

File: src/org/folio/Constants.groovy [hunk 3]
```diff
@@ -229,4 +231,11 @@ class Constants {
     static String DM_ISSUES_JQL = 'labels = dataMigrationSchemaDiff and status != Closed'
 
     static String GLOBAL_BUILD_TIMEOUT = '4'
+
+    static List KitFoxMembers = ["oleksii_petrenko1@epam.com",
```
[the remark this answers is not in the document]
Comment (eldiiar-duishenaliev) [reply src/org/folio/Constants.groovy]: Very valuable, thanks!

File: vars/folioJiraFlow.groovy (hunk 1 above)
Comment (eldiiar-duishenaliev) [reply vars/folioJiraFlow.groovy]: Thanks Sasha, indeed, instead tons of initializations inside loop, better to initialize it once.

**Review by vkapylou — APPROVED** [review]

**Review by guram-jalaghonia — APPROVED** [review]

File: vars/folioJiraFlowMail.groovy [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Build Status: ${currentBuild.currentResult}
+Build Number: ${currentBuild.number}
+Build URL: ${BUILD_URL}
+  """
+  emails.each { mail ->
+    try {
+      emailext body: """${pipelineInfo}\n${body}""", subject: "${subject}", to: "${mail}"
+      new Common(this, "https://fakeUrl").logger.info("Mail sent to: ${mail}")
+    } catch (Exception e) {
+      new Common(this, "https://fakeUrl").logger.warning("Unable to send email to: ${mail}\nError: ${e.getMessage()}")
```
Comment (OHaimanov) [inline vars/folioJiraFlowMail.groovy:16]: As said before better to use Logger class instead of Common

**Review by OHaimanov — APPROVED** [review]

File: vars/folioJiraFlowMail.groovy (hunk 4 above)
Comment (eldiiar-duishenaliev) [reply vars/folioJiraFlowMail.groovy]: Thanks Sasha, somehow, I've missed this.

**Pull request merged**